packages/theme/pages/MyAccount.vue

```diff
@@ -62,56 +47,21 @@ export default {
     OrderHistory,
     OrderDetails
   },
-  middleware: ['is-authenticated'],
+  middleware: ['is-authenticated', 'my-profile'],
   setup(props, context) {
-    const route = useRoute();
     const router = useRouter();
     const { logout } = useUser();
-    const isMobile = computed(mapMobileObserver().isMobile);
-
-    const handleOpenPage = async ({ pageName }) => {
-      const pageRoute = {
-        ...route.value,
-        path: context.root.localePath(route.value.path),
-        params: { pageName }
-      };
-      router.push(pageRoute);
-    };
 
     const handleLogout = async () =>
       logout()
         .then(() => context.root.localePath({ name: 'home' }))
         .then(path => router.push(path));
```

Comment:
I believe you could replace `router.push` with `router.replace`, because the user is unauthorized to go back anyways.

Additionally, this could be refactored to await (instead of promises) to be consistent with the project's code style.
```suggestion
    const handleLogout = async () => {
      await logout();
      router.replace(context.root.localePath({ name: 'home' }));
    };
```